plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeGcsConnectorSmokeTest.java

```
@@ -0,0 +1,294 @@
+                String fileName = resourceInfo.getResourceName().replaceFirst("^" + Pattern.quote(resourcePath), quoteReplacement(targetDirectory));
+                ByteSource byteSource = resourceInfo.asByteSource();
+                try (FSDataOutputStream fileStream = fileSystem.create(new org.apache.hadoop.fs.Path(fileName), true)) {
+                    ByteStreams.copy(byteSource.openBufferedStream(), fileStream);
+                }
+            }
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+
+        queryRunner.execute(format("CREATE TABLE %s (dummy int) WITH (location = '%s')", table, getLocationForTable(bucketName, table)));
+    }
+
+    @Override
+    protected String getLocationForTable(String bucketName, String tableName)
+    {
+        return bucketUrl() + tableName;
+    }
+
+    @Override
+    protected List<String> getTableFiles(String tableName)
+    {
+        return listAllFilesRecursive(tableName);
+    }
+
+    @Override
+    protected List<String> listCheckpointFiles(String transactionLogDirectory)
+    {
+        return listAllFilesRecursive(transactionLogDirectory).stream()
+                .filter(path -> path.contains("checkpoint.parquet"))
+                .collect(toImmutableList());
+    }
+
+    private List<String> listAllFilesRecursive(String directory)
+    {
+        String path = bucketUrl() + directory;
+        ImmutableList.Builder<String> paths = ImmutableList.builder();
+
+        try {
+            RemoteIterator<LocatedFileStatus> files = fileSystem.listFiles(new org.apache.hadoop.fs.Path(path), true);
+            while (files.hasNext()) {
+                LocatedFileStatus file = files.next();
+                if (!file.isDirectory()) {
+                    paths.add(file.getPath().toString());
+                }
+            }
+        }
+        catch (FileNotFoundException e) {
+            return ImmutableList.of();
```

why not fail in case that the directory is not present?

---

There are tests that rely on this behavior. Directory `xyz` doesn't exist, and the test checks how many files are in `xyz`. Seems like a reasonable contract to me